components/match2/wikis/starcraft2/legacy/match_maps_team_legacy.lua

```diff
@@ -0,0 +1,184 @@
+	storageArgs.bestof = args.bestof
+
+	MatchMapsTeamLegacy._handleMaps()
+
+	MatchMapsTeamLegacy._handleOpponents()
+
+	if args.date then
+		args.dateheader = true
+	end
+
+	Template.stashReturnValue(Table.merge(args, storageArgs), 'LegacyMatchlist')
+end
+
+function MatchMapsTeamLegacy._handleMaps()
+	local storageArgs = _storageArgs
+
+	local gameIndex = 1
+	local prefix = 'm' .. gameIndex
+	local map = storageArgs[prefix .. 'map']
+	local mapWinner = storageArgs[prefix .. 'win']
+
+	while map or mapWinner do
+		storageArgs['map' .. gameIndex] = MatchMapsTeamLegacy._processSingleMap(prefix, map, mapWinner)
+
+		gameIndex = gameIndex + 1
+		prefix = 'm' .. gameIndex
+		map = storageArgs[prefix .. 'map']
+		mapWinner = storageArgs[prefix .. 'win']
+	end
+
+	prefix = 'ace'
+	map = storageArgs[prefix .. 'map']
+	mapWinner = storageArgs[prefix .. 'win']
+
+	if map or mapWinner then
+		storageArgs['map' .. gameIndex] = MatchMapsTeamLegacy._processSingleMap(prefix, map, mapWinner)
+	end
+end
+
+function MatchMapsTeamLegacy._processSingleMap(prefix, map, mapWinner)
+	local storageArgs = _storageArgs
+
+	local archon = Logic.readBool(storageArgs[prefix .. 'archon'])
+
+	local mapArgs = {
+		map = map or 'unknown',
+		winner = mapWinner,
+	}
+	mapArgs = MatchMapsTeamLegacy._processMapOpponent(1, prefix, mapArgs, archon)
+	mapArgs = MatchMapsTeamLegacy._processMapOpponent(2, prefix, mapArgs, archon)
+
+	MatchMapsTeamLegacy._removeProcessedMapInput(prefix)
+
+	return mapArgs
+end
+
+function MatchMapsTeamLegacy._processMapOpponent(side, prefix, mapArgs, archon)
+	local storageArgs = _storageArgs
+
+	local addToMapArgs = {
+		['t' .. side .. 'p1'] = String.isNotEmpty(storageArgs[prefix .. 'p' .. side .. 'link'])
+			and storageArgs[prefix .. 'p' .. side .. 'link']
+			or storageArgs[prefix .. 'p' .. side],
+		['t' .. side .. 'p1race'] = storageArgs[prefix .. 'p' .. side .. 'race'],
+		['t' .. side .. 'p1flag'] = storageArgs[prefix .. 'p' .. side .. 'flag'],
+
+		['opponent' .. side .. 'archon'] = archon and 'true' or nil,
+		['opponent' .. side .. 'race'] = archon and storageArgs[prefix .. 'p' .. side .. 'race'] or nil,
+
+		['t' .. side .. 'p2'] = String.isNotEmpty(storageArgs[prefix .. 't' .. side .. 'p2link'])
+			and storageArgs[prefix .. 't' .. side .. 'p2link']
+			or storageArgs[prefix .. 't' .. side .. 'p2'],
+		['t' .. side .. 'p2race'] = storageArgs[prefix .. 't' .. side .. 'p2race'],
+		['t' .. side .. 'p2flag'] = storageArgs[prefix .. 't' .. side .. 'p2flag'],
+
+		['t' .. side .. 'p3'] = String.isNotEmpty(storageArgs[prefix .. 't' .. side .. 'p3link'])
+			and storageArgs[prefix .. 't' .. side .. 'p3link']
+			or storageArgs[prefix .. 't' .. side .. 'p3'],
+		['t' .. side .. 'p3race'] = storageArgs[prefix .. 't' .. side .. 'p3race'],
+		['t' .. side .. 'p3flag'] = storageArgs[prefix .. 't' .. side .. 'p2flag'],
+	}
+
+	MatchMapsTeamLegacy._setPlayersForOpponents(addToMapArgs, side, storageArgs[prefix .. 'p' .. side])
```

Comment:
Why is this called from here and not _handleOpponents?

Reply:
because the data needed is set in the params of the maps and not of the opponents

if i were to do it in the opponent handling i had to loop through everything again (and couldn't remove the processed orig input) as well as having duplicate code (not really duplicate code because it would be added to different tables and with different keys, but the stuff that gets input would be duplicate code)

Reply:
Because of for the storage of the table, so effectively only the last handled map is stored into the table, right?

Reply:
the currently handled map is added to it,
if the player already exists in the table the entry for that it is overwritten
but other players are not overwritten

Reply:
Hm, not a fan of it, but it's legacy and I don't feel like the legacy code needs to of the same code quality as the core code. So I'll accept it